### PR TITLE
format component and function 

### PR DESCRIPTION
**Pull request by JerryWu1234**
#2585 

image
image

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [x] I've updated the relevant documentation for any new or updated feature
- [x] I've linked relevant GitHub issue with "Closes #<issue id>"
- [x] I've added a visual demonstration under the form of a screenshot or video

### CODE REVIEW

File: packages/toolpad-app/src/server/FunctionsManager.ts [hunk 1]
```diff
@@ -19,16 +19,20 @@ import { Awaitable } from '../utils/types';
 import { format } from '../utils/prettier';
 import { compilerOptions } from './functionsShared';
 
-function createDefaultFunction(): string {
-  return format(`
+async function createDefaultFunction(filePath: string) {
```
Comment (Janpot) [inline packages/toolpad-app/src/server/FunctionsManager.ts:22]: To keep the return type for async functions you can always use `: Promise<string>`
```suggestion
async function createDefaultFunction(filePath: string): Promise<string> {
```

**Comment by JerryWu1234** [comment]
@Janpot does your review finish? Anything else ?

**Review by Janpot — APPROVED** [review]
Thank you, this is looking good! If you're up to it can you bring back the return types?

**Comment by JerryWu1234** [comment]
> 

why not .
Let me fix it tomorrow

**Comment by JerryWu1234** [comment]
> Thank you, this is looking good! If you're up to it can you bring back the return types?

Last PR you told me that you also want to format the function and the component.

And tell me any demand please tell me

File: packages/toolpad-app/src/server/FunctionsManager.ts (hunk 1 above)
Comment (JerryWu1234) [reply packages/toolpad-app/src/server/FunctionsManager.ts]: I done

**Review by Janpot — APPROVED** [review]
Thank you, this is great! I've removed the `tryFormat` function as we don't use it anymore.

I've added one more item in https://github.com/mui/mui-toolpad/issues/2585. The bindings editor [formats expressions](https://github.com/mui/mui-toolpad/blob/301e2c1b4d47eb0e69a03ff3c1b69f9f8187395e/packages/toolpad-app/src/toolpad/AppEditor/BindingEditor.tsx#L509), this is a bit more challenging since we run this in the browser, we'll have to read the config on the server and pass it to the frontend.

**Comment by JerryWu1234** [comment]
> Thank you, this is great! I've removed the `tryFormat` function as we don't use it anymore.
> 
> I've added one more item in #2585. The bindings editor [formats expressions](https://github.com/mui/mui-toolpad/blob/301e2c1b4d47eb0e69a03ff3c1b69f9f8187395e/packages/toolpad-app/src/toolpad/AppEditor/BindingEditor.tsx#L509), this is a bit more challenging since we run this in the browser, we'll have to read the config on the server and pass it to the frontend.

ok I will continue to follow up on this issue

**Comment by JerryWu1234** [comment]
> Thank you, this is great! I've removed the `tryFormat` function as we don't use it anymore.
> 
> I've added one more item in #2585. The bindings editor [formats expressions](https://github.com/mui/mui-toolpad/blob/301e2c1b4d47eb0e69a03ff3c1b69f9f8187395e/packages/toolpad-app/src/toolpad/AppEditor/BindingEditor.tsx#L509), this is a bit more challenging since we run this in the browser, we'll have to read the config on the server and pass it to the frontend.

@Janpot what way do we pass it to the frontend ?  Could I create a new API or anything else？
do you have any demand?

**Comment by Janpot** [comment]
You could add a `getPrettierConfig` method in [the project](https://github.com/mui/mui-toolpad/blob/4934af201b65a99724b0b454da6d975b83780b63/packages/toolpad-app/src/server/localMode.ts#L965). No need to specify a file, just resolve the prettier config at the [project root](https://github.com/mui/mui-toolpad/blob/4934af201b65a99724b0b454da6d975b83780b63/packages/toolpad-app/src/server/localMode.ts#L966). Pass it to the frontend and use it the same way we do with [`getVersionInfo`](https://github.com/mui/mui-toolpad/blob/01e3cc61318ac5e9c42380920b5bb1f405c45721/packages/toolpad-app/src/server/rpcServer.ts#L19).

**Pull request merged**